src/selectedtests/task_mappings/mappings.py

```diff
@@ -62,57 +64,66 @@ def create_task_mappings(
             before_date=before_date,
         )
         log.info("Starting to generate task mappings")
-        project_versions: Generator[Version] = evg_api.versions_by_project(evergreen_project)
+        project_versions = evg_api.versions_by_project_time_window(
+            evergreen_project, before_date, after_date
+        )
 
         task_mappings = {}
 
-        base_repo: Repo = None
         module_repo: Repo = None
-        branch = ""
-        repo_name = ""
+        branch = None
+        repo_name = None
 
         with TemporaryDirectory() as temp_dir:
-            for next_version, version, prev_version in windowed_iter(project_versions, 3):
-                if version.create_time < after_date:
-                    break
-                if version.create_time > before_date:
-                    continue
-                if base_repo is None:
+            try:
+                base_repo = _get_evg_project_and_init_repo(evg_api, evergreen_project, temp_dir)
+            except ValueError:
+                LOGGER.warning("Unexpected exception", exc_info=True)
+                raise
+
+            jobs = []
```

Comment:
This is the start of the concurrency code. It is uses futures. The `jobs` variable will keep track of any code we want to execute in the background as futures.